File: AlgoExpert/validate-subsequence.go
package main

// O(n)
func IsValidSubsequence(array []int, sequence []int) bool {
	i := 0
	j := 0
	for i < len(array) && j < len(sequence) {
		if array[i] == sequence[j] {
			j++
		}
		i++
	}
	return j == len(sequence)
}

File: AlgoExpert/validate-subsequence.ts
export function isValidSubsequence(array: number[], sequence: number[]) {
  let sequencePointer = 0;
  for (let i = 0; i < array.length; i++) {
    if (sequence[sequencePointer] === array[i]) {
      sequencePointer++;
    }
  }
  return sequencePointer === sequence.length;
}
